### PR TITLE
Fix system image, add test

### LINKED ISSUE #380: System image fails with Julia 1.7.0 and new versions of the dependencies
We identified a StackOverflowError in the creation of the system image recently, due to the dependencies of CalibrateEDMF. The stack tries to run indefinitely:

`isLoadFromConstGV at /buildworker/worker/package_linux64/build/src/llvm-late-gc-lowering.cpp:1153`

This has already been reported by other users in Discourse: https://discourse.julialang.org/t/package-compiler-fails-when-using-julia-1-7/72421

Originally found by @haakon-e

### PR DESCRIPTION
**Pull request by ilopezgp**
## Changes

- Fixes system image, which stopped working on Julia 1.7.0 due to issue https://github.com/JuliaLang/PackageCompiler.jl/issues/643.
- Adds system image test to buildkite, created by @haakon-e.
- Updates all scripts, tests and docs to Julia 1.7.3.

Co-authored by @haakon-e.

## Issue number (if applicable)

Closes #380 

## Checklist before requesting a review / merging
- [x] I have formatted the code using `julia --project=.dev .dev/climaformat.jl .`
- [x] I have updated all test manifests using `julia --project .dev/up_deps.jl .` with Julia 1.7.3.
- [x] If core features are added, I have added appropriate test coverage.
- [x] If new functions and structs are added, they are documented through docstrings.

### CODE REVIEW

**Comment by ilopezgp** [comment]
`pkgs = [:TurbulenceConvection]` works.

**Comment by haakon-e** [comment]
With this, I will close #378, #361

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -44,6 +44,16 @@ steps:
...
+  - label: "create system image"  
+    key: "cpu_create_so"
+    command:
+      - "julia --project=integration_tests -C skylake-avx512 integration_tests/test_create_tc_so.jl"
+    agents:
+      config: cpu
+      queue: central
+      slurm_cpus_per_task: 1
+      slurm_nodes: 1
+      slurm_mem_per_cpu: 30G
```
Comment (haakon-e) [inline .buildkite/pipeline.yml:56]: we may not need 30G of memory for this. The relevant slurm scripts for calibration only requests 20G. I have also been able to generate a sys image (note: I didn't run `test`) with only 4G.

File: integration_tests/SCT1_integration_test_config.jl [hunk 2]
```diff
@@ -24,6 +24,7 @@ namelist_args = [
     ("time_stepping", "dt_max", 2.0),
     ("stats_io", "frequency", 60.0),
     ("stats_io", "calibrate_io", true),
+    ("thermodynamics", "sgs", "mean"),
```
[the remark this answers is not in the document]
Comment (ilopezgp) [reply integration_tests/SCT1_integration_test_config.jl:27]: The SCT1 integration test is failing for some reason, I am trying to see if it was the change of thermodynamics.

**Comment by haakon-e** [comment]
It seems you also need to update the julia version in:
* `run_tests.sbatch`
* `tools/TCRunner.sbatch`

**Comment by ilopezgp** [comment]
> It seems you also need to update the julia version in:
> 
> * `run_tests.sbatch`
> * `tools/TCRunner.sbatch`

thanks! I missed those.

**Review by haakon-e — APPROVED** [review]
great job!

**Comment by ilopezgp** [comment]
> great job!

Same! 🥇

**Pull request merged**